applications/GeoMechanicsApplication/custom_elements/updated_lagrangian_U_Pw_diff_order_element.cpp

```diff
@@ -48,100 +45,41 @@ void UpdatedLagrangianUPwDiffOrderElement::CalculateAll(MatrixType&        rLeft
-            IntegrationPoints[GPoint], Variables.detJInitialConfiguration);
-
-        if (CalculateStiffnessMatrixFlag) {
-            // Contributions to stiffness matrix calculated on the reference config
-            /* Material stiffness matrix */
-            this->CalculateAndAddLHS(rLeftHandSideMatrix, Variables);
-
-            /* Geometric stiffness matrix */
-            if (Variables.ConsiderGeometricStiffness)
-                this->CalculateAndAddGeometricStiffnessMatrix(rLeftHandSideMatrix, Variables, GPoint);
-        }
+    SmallStrainUPwDiffOrderElement::CalculateAll(rLeftHandSideMatrix, rRightHandSideVector,
+                                                 rCurrentProcessInfo, CalculateStiffnessMatrixFlag,
+                                                 CalculateResidualVectorFlag);
+
+    ElementVariables variables;
+    this->InitializeElementVariables(variables, rCurrentProcessInfo);
+
+    if (CalculateStiffnessMatrixFlag && variables.ConsiderGeometricStiffness) {
+        const auto& integration_points = this->GetGeometry().IntegrationPoints(this->GetIntegrationMethod());
+        const auto integration_coefficients =
+            this->CalculateIntegrationCoefficients(integration_points, variables.detJuContainer);
 
-        if (CalculateResidualVectorFlag) {
-            // Contributions to the right hand side
-            this->CalculateAndAddRHS(rRightHandSideVector, Variables, GPoint);
+        for (IndexType GPoint = 0; GPoint < integration_points.size(); ++GPoint) {
+            this->CalculateAndAddGeometricStiffnessMatrix(rLeftHandSideMatrix, mStressVector[GPoint],
+                                                          variables.DNu_DXContainer[GPoint],
+                                                          integration_coefficients[GPoint]);
         }
     }
 
     KRATOS_CATCH("")
 }
 
-//----------------------------------------------------------------------------------------
 void UpdatedLagrangianUPwDiffOrderElement::CalculateAndAddGeometricStiffnessMatrix(
-    MatrixType& rLeftHandSideMatrix, const ElementVariables& rVariables, unsigned int GPoint)
+    MatrixType& rLeftHandSideMatrix, const Vector& rStressVector, const Matrix& rDNuDx, const double IntegrationCoefficient)
 {
     KRATOS_TRY
 
     const GeometryType& r_geom      = GetGeometry();
     const SizeType      num_U_nodes = r_geom.PointsNumber();
     const SizeType      dimension   = r_geom.WorkingSpaceDimension();
 
-    const auto stress_tensor = MathUtils<double>::StressVectorToTensor(mStressVector[GPoint]);
+    const auto stress_tensor = MathUtils<double>::StressVectorToTensor(rStressVector);
 
     const Matrix reduced_Kg_matrix =
-        prod(rVariables.DNu_DX,
-             rVariables.IntegrationCoefficient * Matrix(prod(stress_tensor, trans(rVariables.DNu_DX)))); // to be optimized
+        prod(rDNuDx, IntegrationCoefficient * Matrix(prod(stress_tensor, trans(rDNuDx))));

```

Comment:
Place of IntegrationCoefficient multiplication and intermediate variable. See the other comment.

Reply:
Changed